src/main/java/com/aws/greengrass/mqttclient/MqttClient.java

```diff
@@ -979,6 +936,65 @@ protected int getNextClientIdNumber() {
         return 0;
     }
 
+    /**
+     * Create AwsIotMqttConnectionBuilder to generate an mqtt connection client.
+     *
+     * @param deviceConfiguration used to get the values needed for mqtt connection builder
+     * @param clientBootstrap client bootstrap values for mqtt connection builder
+     */
+    @SuppressWarnings("PMD.PreserveStackTrace")
+    public AwsIotMqttConnectionBuilder createMqttConnectionBuilder(DeviceConfiguration deviceConfiguration,
+                                                                   ClientBootstrap clientBootstrap) {
+        AwsIotMqttConnectionBuilder builder;
+        try {
+            builder = securityService.getDeviceIdentityMqttConnectionBuilder();
```

Comment:
this will get the security service for the existing configuration

Reply:
Is there a reason we can’t use the existing security service? It gets the private key and cert file path from the existing configuration and that’s not something that can be changed with a deployment

Reply:
I am pointing it out. Please add a comment in the code to point it out too.